docs/adr/0004-fargate-pools.md

```diff
@@ -0,0 +1,44 @@
+# 4. EKS FargatePools
+
+- Status: proposed
+- Date: 2021-02-15
+- Authors: @michaelbeaumont
+- Deciders:
+
+## Context
+
+At the time of writing CAPA has the AWS-specific infrastructure resources `AWSMachinePool`s
+and `AWSManagedMachinePool`s.
+These are referred to in `MachinePool`s as `infrastructureRef`s. The
+underlying AWS resources fit the `MachinePool` model very closely. Data about
+the pools, such as `replicas` is reflected back to the `MachinePool` and
+the responsible controllers reconcile the AWS resources using data specified on the
+`MachinePool`, like `.spec.template.spec.version` and `.spec.template.spec.bootstrap`.
+
+Issue #1786 is concerned with supporting EKS Fargate profiles in CAPA. Fargate
+profiles allow users to run pods on an on-demand, serverless compute platform
+called Fargate.
+Nodes are spun up dynamically for pods in specified namespaces or with specific
+labels. These selectors are packaged into a "profile" that also specifies which
+subnets are available for pods to run in.
+
+With CAPA, users would like to create a resource representing such a profile.
+
+This ADR concerns itself with how these resources are specified. The major
+decision is whether a Fargate profile can be represented in the `MachinePool` abstraction.
+
+Some of the fields on `MachinePool` don't really make sense in the Fargate
+context, like the entire `MachineSpec` under `spec.template.spec` (`version`,
+`bootstrap`), `spec.strategy` and the idea of desired replicas.
+
+## Decision
+
+The `FargatePool` resource is implemented independently from any existing CAPI resource 
+and users must only create and manipulate `FargatePool`s to create and manage
+Fargate profiles.
+
```

Comment:
Is it worth mentioning that even though we aren't using the MachinePool CAPI kind that the AWSFaregatePool will still be associated with a Cluster/AWSManagedControlPlane?

Reply:
Thanks